pkg/charts/utils.go

```diff
@@ -297,6 +304,14 @@ func generateChartValues(config *ciliumv1alpha1.NetworkConfig, network *extensio
 		globalConfig.BGPControlPlane.Enabled = config.BGPControlPlane.Enabled
 	}
 
+	if config.Encryption != nil && config.Encryption.Enabled {
+		globalConfig.Encryption.Enabled = config.Encryption.Enabled
+		globalConfig.Encryption.KeyFile = config.Encryption.KeyFile
```

Comment:
Here you are overwriting the defaults you are setting above. This is probably not intended.

Reply:
Hi @axel7born true, fixed it